executor/adapter.go

```diff
@@ -1257,13 +1257,18 @@ func (a *ExecStmt) observeStmtBeginForTopSQL() {
 		vars.StmtStats.OnExecutionBegin(sqlDigest, planDigest)
 		// This is a special logic prepared for TiKV's SQLExecCount.
 		vars.StmtCtx.KvExecCounter = vars.StmtStats.CreateKvExecCounter(sqlDigest, planDigest)
+		failpoint.Inject("mockSleepInExec", func(ms failpoint.Value) {
+			time.Sleep(time.Duration(ms.(int)) * time.Millisecond)
+
+		})
```

Comment:
Why not simply use a sql with `select sleep(1)` or something? The failpoint here doesn't prevent you forget to call `observeStmtBeginForTopSQL` or call it in wrong place, result in not very effective test cases.